subsys/shell/modules/kernel_service.c

```diff
@@ -100,14 +111,13 @@ static void shell_stack_dump(const struct k_thread *thread, void *user_data)
 	const char *tname;
 
 	tname = k_thread_name_get((struct k_thread *)thread);
-	unused = stack_unused_space_get((char *)thread->stack_info.start,
-					size);
+	unused = stack_unused_space_get((char *)thread->stack_info.start, size);
 
 	/* Calculate the real size reserved for the stack */
 	pcnt = ((size - unused) * 100U) / size;
 
-	shell_fprintf((const struct shell *)user_data, SHELL_NORMAL,
-		"0x%08X %-10s (real size %u):\tunused %u\tusage %u / %u (%u %%)\n",
+	shell_print((const struct shell *)user_data,
+		"0x%08X %-10s (real size %u):\tunused %u\tusage %u / %u (%u %%)",
```

Comment:
Just a hint. If you do not need fixed width `-10` you can use variable length with `*`.